deploy/FullDeploy.js

```diff
@@ -45,29 +45,23 @@ module.exports = async function (hre) {
     })
 
     // deploy oracles
-    const priceOracle = await deploy("PriceOracle", {
+    const gasOracle = await deploy("GasOracle", {
         from: deployer,
         log: true,
         contract: "Oracle",
     })
 
-    const oracleAdapter = await deploy("PriceOracleAdapter", {
-        from: deployer,
-        log: true,
-        args: [priceOracle.address],
-        contract: "OracleAdapter",
-    })
-
-    const gasOracle = await deploy("GasOracle", {
+    const ethOracle = await deploy("EthOracle", {
         from: deployer,
         log: true,
         contract: "Oracle",
     })
 
-    const ethOracle = await deploy("EthOracle", {
+    const oracleAdapter = await deploy("PriceOracleAdapter", {
         from: deployer,
         log: true,
-        contract: "Oracle",
+        args: [ethOracle.address],
+        contract: "OracleAdapter",
     })
```

Comment:
This is using ethOracle, but it should be using priceOracle, which you've deleted.

price Oracle is for the price of the base asset, eth oracle is for ETH/USD (used for calculating liquidation gas cost)

Reply:
is see, but this is working now. Should the priceOracle be passed into the Tracer contract??